pymobiledevice3/services/device_link.py

```diff
@@ -87,10 +88,19 @@ def download_files(self, message):
                 self.service.sendall(struct.pack(SIZE_FORMAT, len(e.strerror) + struct.calcsize(CODE_FORMAT)))
                 self.service.sendall(struct.pack(CODE_FORMAT, CODE_ERROR_LOCAL) + e.strerror.encode())
             else:
-                self.service.sendall(struct.pack(SIZE_FORMAT, len(data) + struct.calcsize(CODE_FORMAT)))
-                self.service.sendall(struct.pack(CODE_FORMAT, CODE_FILE_DATA) + data)
+                # split into chunks, otherwise we may crash BackupAgent2 by OOM
```

Comment:
let's change the `read_bytes()` to instead read from the file in chunks so its not save in whole onto memory

Reply:
Resolved.